server/tests/ballot_comparison/test_cvrs.py

```diff
@@ -2169,7 +2149,169 @@ class TestCase(TypedDict):
+            "files": [
+                (
+                    zip_hart_cvrs(HART_CVRS_DUPLICATE_BATCH_NAMES["TABULATOR1"]),
+                    "TABULATOR1.zip",
+                ),
+                (
+                    zip_hart_cvrs(HART_CVRS_DUPLICATE_BATCH_NAMES["TABULATOR2"]),
+                    "TABULATOR2.zip",
+                ),
+            ],
+            "expected_processing_status": ProcessingStatus.PROCESSED,
+            "expected_processing_error": None,
+        },
+        {
+            # Failing to extract tabulator info
+            "files": [
+                (
+                    zip_hart_cvrs(
+                        [
+                            *HART_CVRS_DUPLICATE_BATCH_NAMES["TABULATOR1"],
+                            *HART_CVRS_DUPLICATE_BATCH_NAMES["TABULATOR2"],
+                        ]
+                    ),
+                    "cvrs.zip",
+                ),
+            ],
+            "expected_processing_status": ProcessingStatus.ERRORED,
+            "expected_processing_error": "Couldn't find a tabulator for CVR 1-1-1. Because the batch names in your ballot manifest are not unique, CVR tabulator info is needed. These can be provided by uploading scanned ballot information CSVs or a CVR ZIP file per tabulator, where the ZIP file names are tabulator names.",
+        },
+    ]
+
+    for test_case in test_cases:
+        rv = client.put(
+            f"/api/election/{election_id}/jurisdiction/{jurisdiction_ids[0]}/cvrs",
+            data={"cvrs": test_case["files"], "cvrFileType": "HART",},
+        )
+        assert_ok(rv)
+
+        rv = client.get(
+            f"/api/election/{election_id}/jurisdiction/{jurisdiction_ids[0]}/cvrs"
+        )
+        compare_json(
+            json.loads(rv.data),
+            {
+                "file": {
+                    "cvrFileType": "HART",
+                    "name": "cvr-files.zip",
+                    "uploadedAt": assert_is_date,
+                },
+                "processing": {
+                    "status": test_case["expected_processing_status"],
+                    "startedAt": assert_is_date,
+                    "completedAt": assert_is_date,
+                    "error": test_case["expected_processing_error"],
+                    "workProgress": (
+                        0
+                        if test_case["expected_processing_status"]
+                        == ProcessingStatus.ERRORED
+                        else manifest_num_ballots
+                    ),
+                    "workTotal": manifest_num_ballots,
+                },
+            },
+        )
+
+        if test_case["expected_processing_status"] == ProcessingStatus.PROCESSED:
+            cvr_ballots = (
+                CvrBallot.query.join(Batch)
+                .filter_by(jurisdiction_id=jurisdiction_ids[0])
+                .order_by(CvrBallot.imprinted_id)
+                .all()
+            )
```

Comment:
I know we'd rather use the API layer for these checks, instead of retrieving records from the DB directly. Just mimicking the convention used by other tests in this file